tracks/rust/exercises/bob/mentoring.md

```diff
@@ -106,11 +93,6 @@ Then note the following:
 See about finding a way to check if it's alphabetic without using regex.
 ```
 
-If they use separate functions:
```

Comment:
I see you say 

> It's recommended to use functions paths instead of closures where possible.

Is this a _language recommendation_? Or a personal one? From what I've seen from my own fiddling with rust is that in production-projects closures are used heavily and I can't seem to match this recommendation. Some science here will probably really help with accepting this change, given that it doesn't clash with `track-anatomy`.

Reply:
`foo` is simply shorter than `|x| x.foo()`. Clippy suggests to remove redundant closures as well.

Reply:
Oh, I see this is about something else.

The mentoring notes originally said that you should _not_ use helper functions, but I disagree. This is a subject on which there are many different opinions. We shouldn't force students to use a certain pattern, unless there's a _clear_ advantage.

> It's recommended to use functions paths instead of closures where possible.

I should have phrased this better :)

This is about closures like `|c: char| c.is_alphabetic()`, which can be replaced with `char::is_alphabetic`.

The [clippy source](https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/eta_reduction.rs#L18) says about this: “Needlessly creating a closure adds code for no benefit and gives the optimizer more work.”